### PR TITLE
Add test for zero amount payment_v2 txns

### PR DESCRIPTION
**Pull request by vihu**
Related to: https://github.com/helium/blockchain-core/pull/421

### CODE REVIEW

File: test/miner_payment_v2_txn_SUITE.erl [hunk 1]
```diff
@@ -137,3 +140,64 @@ basic_test(Config) ->
...
+    PayeeAmount = 0,
+    Payments = [blockchain_payment_v2:new(P, PayeeAmount) || P <- PayeeAddrs],
+    Txn = ct_rpc:call(Payer, blockchain_txn_payment_v2, new, [PayerAddr, Payments, 1, Fee]),
+
+    {ok, _Pubkey, SigFun, _ECDHFun} = ct_rpc:call(Payer, blockchain_swarm, keys, []),
+
+    SignedTxn = ct_rpc:call(Payer, blockchain_txn_payment_v2, sign, [Txn, SigFun]),
+    ct:pal("SignedTxn: ~p", [SignedTxn]),
+
+    {[], [SignedTxn]} = ct_rpc:call(Payer, blockchain_txn, validate, [[SignedTxn], Chain]),
```
Comment (andymck) [inline test/miner_payment_v2_txn_SUITE.erl:183]: Kinda a nit but maybe better to do something like ?assertEqual(false, TXN_TYPE:is_valid(Txn, Chain)) ?
Make the validation check more explicit and obvious

Comment (vihu) [reply test/miner_payment_v2_txn_SUITE.erl]: Certainly, added d149bca
